src/base/bittorrent/session.cpp

```diff
@@ -2413,13 +2413,30 @@ void Session::networkConfigurationChange(const QNetworkConfiguration& cfg)
     }
 }
 
+bool Session::addressPatternMatches(const QString& ipAddress, const QString& pattern)
+{
+    const QStringList ipPieces = ipAddress.split(".");
+    const QStringList patternPieces = pattern.split(".");
+
+    if (ipPieces.size() != patternPieces.size())
+        return false;
+
+    for (int i = 0; i < ipPieces.size(); i++)
+        if (patternPieces[i] != "*" && ipPieces[i] != patternPieces[i])
```

Comment:
Please wrap each part of the conditional in parentheses

`if ((patternPieces[i] != "*") && (ipPieces[i] != patternPieces[i]))`